### ISSUE TITLE
Display As does not uniquely identify tokens

### ISSUE THREAD

**Issue by xusheng6** [Vector35/binaryninja-api#3490]
The implementation of "Display As" is differentiating tokens based on their values. This is problematic -- for example, if we have two different tokens that happen to have the same value at the same address, then setting the display type of one of them will also set the display type of the other one.

For the following code
```
  30 @ 100003ec0  if (temp4 u> 0x22) then 34 @ 0x100003ed8 else 49 @ 0x100003ec4
```
After I set the display type of either the 0x22 or 34 token (0x22 == 34), I get this
```
100003ec0  if (temp4 u> 4.76441478e-44) then 4.76441478e-44 @ 0x100003ed8 else 49 @ 0x100003ec4
```

The display as use the tuple of (value, operand_index) to identify tokens. I am not sure whether the operand_index is supposed to be different for the two tokens, but involving the value of the token is definitely not a good idea and a source of obscure bugs

**Comment by xusheng6** [comment]
This affects the IL views the most. Since in disassembly, the `operand` somehow uniquely identifies the tokens in most cases. But in IL views, the operand tends to be 0xffffffff, so it does not help.

**Comment by xusheng6** [comment]
The expr_index of the token might be better. But it also does not uniquely identify things, since we can easily have two constant values in the same expression.

**Comment by CX330Blake** [comment]
For example, the array length token cannot be displayed as decimal here, but the one at `0x4019a9` is correctly displayed (value 256).

![Image](https://github.com/user-attachments/assets/d16a6a36-644d-4128-a49a-154c013c478b)

**Comment by utkonos** [comment]
I just ran into this bug. Here are three immediates that represent `\`. You can see that the display can be set independently in Disassembly view because they are all three on separate lines.

<img width="458" height="202" alt="Image" src="https://github.com/user-attachments/assets/206c682e-e6c7-41ab-a4ba-f8800df55d1f" />

But here in HLIL, if I change the display of one, it changes all three. One cannot be changed independently of the others. For my function here, this is ok because they actually are the same character. But if I needed them to be set independently, it would be impossible.

<img width="719" height="95" alt="Image" src="https://github.com/user-attachments/assets/0d2330c0-a5e3-4da7-988c-a5e7253e937f" />

**Comment by plafosse** [comment]
As a test if you go down to disassembly or LLIL you can control them independently correct?

**Comment by utkonos** [comment]
@plafosse yes, correct. They are independent in disassembly and LLIL. Here is a screenshot of LLIL

<img width="604" height="510" alt="Image" src="https://github.com/user-attachments/assets/f22125d0-3246-4e3e-b08f-4a7e444524ac" />

**Comment by utkonos** [comment]
I found a counter example, however.

<img width="504" height="87" alt="Image" src="https://github.com/user-attachments/assets/9328d56e-072a-4086-97b0-810f917e8103" />

Here the behavior is different. Each of these hex bytes can have display settings changed independently in HLIL, LLIL, and disasm.

<img width="610" height="395" alt="Image" src="https://github.com/user-attachments/assets/7e3f6d2d-4c32-47cb-9dd1-8e973464865f" />

<img width="521" height="439" alt="Image" src="https://github.com/user-attachments/assets/a48c91ba-64dc-47b6-8acd-6196a3e4fdb7" />

**Comment by plafosse** [comment]
So we have to uniquely identify a constant by triple of (value, address, operand). I believe we can fix this in HLIL by getting the llil instruction's address.
